aten/src/ATen/native/CPUBlas.cpp

```diff
@@ -1161,6 +1161,38 @@ void brgemm(
   "BFloat16 Brgemm is currently not supported");
 }
 
+void brgemm(
+    int64_t M,
+    int64_t N,
+    int64_t K,
+    int64_t ld_a,
+    int64_t ld_b,
+    int64_t ld_c,
+    const float alpha,
+    const float beta,
+    const float* A,
+    const float* B,
+    float* C) {
+  TORCH_CHECK(false,
+  "float Brgemm is currently not supported");
```

Comment:
Can you hide the brgemm path behind an `if constexpr (std::is_same_v<scalar_t, at::Half>)` instead of creating these?

Reply:
Thanks for your comments. Use `if constexpr (std::is_same_v<scalar_t, at::Half>)` for brgemm to remove such unsupported functions.